wallet-new/server/Cardano/Wallet/API/V1/Swagger.hs

```diff
@@ -231,150 +181,223 @@ The number of entries to display for each page. The minimum is **1**, whereas th
+     -H "Accept: application/json; charset=utf-8"                                          \
+     --cacert ./scripts/tls-files/ca.crt                                                   \
+```
 
-This is the specification for the Cardano Wallet API, automatically generated
-as a [Swagger](https://swagger.io/) spec from the [Servant](http://haskell-servant.readthedocs.io/en/stable/) API
-of [Cardano](https://github.com/input-output-hk/cardano-sl).
+Since you have, for now, only a single wallet, you'll see something like this:
+
+```json
+{
+    "status": "success",
+    "data": [
+        {
+            "index": 2147483648,
+            "addresses": [
+                "DdzFFzCqrh...fXSru1pdFE"
+            ],
+            "amount": 0,
+            "name": "Initial account",
+            "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"
+        }
+    ],
+    "meta": {
+        "pagination": {
+            "totalPages": 1,
+            "page": 1,
+            "perPage": 10,
+            "totalEntries": 1
+        }
+    }
+}
+```
 
-Git revision: **$deGitRevision**
-Software version: **$deSoftwareVersion**
+Each account has at least one address, all listed under the `addresses` field. You can
+communicate one of these addresses to receive money on the associated account.
 
 
-## Request format (all versions)
+Sending Money
+=============
 
-Issuing requests against this API is conceptually not very different from any other web service out there. The API
-is **versioned**, meaning that is possible to access different versions of the API by adding the _version number_
-in the URL.
+In order to send money from one of your account to another address, you can create a new
+payment transaction using the `POST /api/v1/transactions` endpoint as follow:
 
-**For the sake of backward compatibility, we expose the legacy version of the API, available simply as
-unversioned endpoints.**
+```
+curl -X POST https://localhost:8090/api/v1/transactions \
+     -H "Content-Type: application/json; charset=utf-8" \
+     -H "Accept: application/json; charset=utf-8"       \
+     --cacert ./scripts/tls-files/ca.crt                \
+     -d '{                                              \
+  "destinations": [{                                    \
+    "amount": 14,                                       \
+    "address": "A7k5bz1QR2...Tx561NNmfF"                \
+  }],                                                   \
+  "source": {                                           \
+    "accountIndex": 0,                                  \
+    "walletId": "Ae2tdPwUPE...8V3AVTnqGZ"               \
+  }                                                     \
+}'
+```
 
-This means that _omitting_ the version number would call the old version of the API. Examples:
+Note that, in order to perform a transaction, you need to have some existing coins on the
+source account! When the transaction succeeds, funds are _moved_ from an address to another.
```

Comment:
A couple of things here:

* We should somehow mention our API supports out-of-the-box multiple recipients payments, which means you can pay more than 1 entity with a single transaction;

* I'm wondering if saying that funds are "moved" is so generic is better to be slightly more specific;

* We should mention that, due to the fact we cannot "preview" a transaction the only way to be aware of the transaction fees for this payment is to call `/api/v1/transactions/fees` which serves exactly this purpose: "previewing" the transaction fees for a payment.

Reply:
For now let's not worry too much about it, but let it marinade. I'm all for embedding informations in the types, but on this particular topic I don't think it would be beneficial. Servant error messages are already quite daunting, and I'm wondering what would happen if we would throw textual documentation into the verbosity 😱

Reply:
Haha ^.^ .. from my own experience, it doesn't get much worse (can it though :/ ?); this is actually the same strategy we use for tagging.

What (other) strategy would recommend to document endpoints' descriptions?

Reply:
> What (other) strategy would recommend to document endpoints' descriptions?

I don't know if that's a stupid idea or even possible, but the next obvious choice would be to do that in our `Swagger.hs` module and use lenses and the traversals provided by `servant-swagger` to annotate the endpoint that way. This would have the advantage we can still use quasiquotes and embed markdown syntax without worrying about breaking GHC 😁

Reply:
I'd suggest bringing this up in #servant channel on freenode, the guys are very responsive and would come up with answers they might already resolve in private projects as well.

Reply:
Digging a bit into servant's issues, I just found a comment referring to: 

https://hackage.haskell.org/package/servant-0.12/docs/Servant-API-Description.html

Looks like what we are looking for :)

Reply:
@domenkozar Yes, that would be a good idea, or even hit /r/haskell for some long-term discussion 😁 

@KtorZ Yep, this is also what I have used in the past but I'm not fully sold that embedding so much info into the types is a good idea here, but I'm happy to be proven wrong 😉

Reply:
At my previous job we'd mostly use `servant-swagger` to generate routes and parameter specs -- the documentation aspect was mostly done to the generated spec, precisely so we could do it at the value level. Stuff like quasiquoters for multiline strings and `mconcat` are very nice and unavailable on symbols :\